### PR TITLE
feat(deps): bump `iam-contracts` version to latest

### PR DESCRIPTION
**Pull request by Harasz**
Update `iam-contracts` package to the latest version, this step require to update `ethers`, `typechain` and `ew-did-registry` as well.

### CODE REVIEW

**Review by jrhender — COMMENTED** [review]
I think it's looking good, just one comment. Let me know when the PR is ready for another review 👍

File: src/modules/staking/staking.service.ts [hunk 1]
```diff
@@ -80,12 +78,9 @@ export class StakingService implements OnModuleDestroy {
     });
   }
   private InitEventListeners(): void {
-    this.stakingPoolFactory.addListener(
-      'StakingPoolLaunched',
-      async (org, address) => {
-        await this.saveStakingPool(address);
-      },
-    );
+    this.stakingPoolFactory.on('StakingPoolLaunched', async (_, address) => {
```
Comment (ahmedolaibrahim) [inline src/modules/staking/staking.service.ts:81]: curious question here:  is `on` equivalent to `addListener`  in the sense that it initialises and adds the listener? if not,  just thinking that we might no longer need the removeListener part.

**Review by ahmedolaibrahim — COMMENTED** [review]
Thanks @Harasz PR looks good overall. just some few clarifying comments.

File: src/modules/staking/staking.service.ts (hunk 1 above)
Comment (Harasz) [reply src/modules/staking/staking.service.ts:81]: There is no longer `addListener` method, just `on` and under the hood it is calling `_addEventListener`. So I think we still need to `removeListener` part.

**Review by ahmedolaibrahim — APPROVED** [review]

**Review by jrhender — APPROVED** [review]
Looks good!

**Pull request merged**